### PR TITLE
Updates based on Paul's secdir review.

### PR DESCRIPTION
**Pull request by chris-wood**

### CODE REVIEW

**Review by csperkins — APPROVED** [review]
No objections on my part.

**Comment by chris-wood** [comment]
@tfpauly, can you please take a look?

File: draft-ietf-taps-transport-security.md [hunk 1]
```diff
@@ -482,30 +485,33 @@ but this document considers them together, since that is the most common pattern
...
 #### IKEv2
 
-IKEv2 is a control protocol that runs on UDP port 500. Its primary goal is to generate keys for Security Associations (SAs).
-An SA contains shared (cryptographic) information used for establishing other SAs or keying ESP; See {{ESP}}.
-IKEv2 first uses a Diffie-Hellman key exchange to generate keys for the "IKE SA", which is a set of keys
-used to encrypt further IKEv2 messages. It then goes through a phase of authentication in which both peers
-present blobs signed by a shared secret or private key, after which another set of keys is derived, referred
-to as the "Child SA". These Child SA keys are used by ESP.
+IKEv2 is a control protocol that runs on UDP port 500 or TCP port 4500. (IKEv2 cannot run over
+UDP on any other port.) Its primary goal is to generate keys for Security Associations (SAs).
```
Comment (tfpauly) [inline draft-ietf-taps-transport-security.md]: Nit on this: I don't like that the entire sentence `(IKEv2 cannot run..` is a parenthetical. Also, while it technically isn't specified to listen on any other port, implementations often in practice use other ports on the initiator end, often because use of 500 or 4500 is restricted.

File: draft-ietf-taps-transport-security.md [hunk 2]
```diff
@@ -482,30 +485,33 @@ but this document considers them together, since that is the most common pattern
 
 #### IKEv2
 
-IKEv2 is a control protocol that runs on UDP port 500. Its primary goal is to generate keys for Security Associations (SAs).
-An SA contains shared (cryptographic) information used for establishing other SAs or keying ESP; See {{ESP}}.
-IKEv2 first uses a Diffie-Hellman key exchange to generate keys for the "IKE SA", which is a set of keys
-used to encrypt further IKEv2 messages. It then goes through a phase of authentication in which both peers
-present blobs signed by a shared secret or private key, after which another set of keys is derived, referred
-to as the "Child SA". These Child SA keys are used by ESP.
+IKEv2 is a control protocol that runs on UDP port 500 or TCP port 4500. (IKEv2 cannot run over
```
Comment (tfpauly) [inline draft-ietf-taps-transport-security.md]: It's UDP ports 500 or 4500 and TCP port 4500

**Review by tfpauly — APPROVED** [review]
Approved with IKE nits

File: draft-ietf-taps-transport-security.md [hunk 3]
```diff
@@ -482,30 +485,33 @@ but this document considers them together, since that is the most common pattern
...
+which is a set of keys used to encrypt further IKEv2 messages. IKE then performs a phase of
+authentication in which both peers present blobs signed by a shared secret or private key that
+authenticates the entire IKE exchange and the IKE identities. IKE then derives further sets of
+keys on demand, which together with traffic policies are referred to as the "Child SA". These
+Child SA keys are used by ESP.
 
 IKEv2 negotiates which protocols are acceptable to each peer for both the IKE and Child SAs using
-"Proposals". Each proposal may contain an encryption algorithm, an authentication algorithm, a
-Diffie-Hellman group, and (for IKE SAs only) a pseudorandom function algorithm. Each peer may
+"Proposals". Each proposal contains an encryption with authentication algorithm or an AEAD algorithm,
```
Comment (squarooticus) [inline draft-ietf-taps-transport-security.md]: Nit: This is a little confusing, maybe since I don't know IPsec all that well: is the "encryption with authentication algorithm" in lieu of AEAD something like AES with HMAC-SHA256, or is an AEAD algorithm required and "or an AEAD algorithm" is an apposition? Either way, I think something needs to be reworded.

File: draft-ietf-taps-transport-security.md [hunk 4]
```diff
@@ -482,30 +485,33 @@ but this document considers them together, since that is the most common pattern
...
 
-Any SA used by IKEv2 can be rekeyed upon expiration, which is usually based either on time or
+Any SA used by IKEv2 can be rekeyed before expiration, which is usually based either on time or
 number of bytes encrypted.
 
 There is an extension to IKEv2 that allows session resumption {{RFC5723}}.
 
 MOBIKE is a Mobility and Multihoming extension to IKEv2 that allows a set of Security Associations
-to migrate over different addresses and interfaces {{RFC4555}}.
+to migrate over different outer IP addresses and interfaces {{RFC4555}}.
```
Comment (squarooticus) [inline draft-ietf-taps-transport-security.md:514]: Should we assume the user knows what an "outer IP address" is? I know there's an informational reference there, but "outer IP address" seems like the answer to a question the user hasn't asked.

**Review by squarooticus — APPROVED** [review]

File: draft-ietf-taps-transport-security.md (hunk 2 above)
Comment (chris-wood) [reply draft-ietf-taps-transport-security.md]: :+1: will fix!

File: draft-ietf-taps-transport-security.md (hunk 1 above)
Comment (chris-wood) [reply draft-ietf-taps-transport-security.md]: Okay, noted. I'll just drop that sentence. It's sort of irrelevant anyway, which is why I made it a parenthetical.

File: draft-ietf-taps-transport-security.md (hunk 4 above)
Comment (chris-wood) [reply draft-ietf-taps-transport-security.md:514]: I assume the interested and uninformed reader would follow the citation to RFC 4555. Should we say more?

File: draft-ietf-taps-transport-security.md (hunk 3 above)
Comment (chris-wood) [reply draft-ietf-taps-transport-security.md]: According to @tfpauly, it's one or the other! Thanks for pointing this out. I'll rephrase it to:

“Each proposal specifies an encryption and authentication algorithm, or an AEAD algorithm, ...”
